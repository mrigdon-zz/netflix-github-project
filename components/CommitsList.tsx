import { useContext } from "react";
import Commit from "../githubAPI/Commit";
import styles from "../styles/CommitsList.module.css";
import { LocaleContext, timestamp } from "../utils/i18n";

function avatarUrl(commit: Commit) {
  const committer = commit.committer || commit.commit.committer;
  return committer.avatar_url;
}

function authorName(commit: Commit) {
  return commit.author ? commit.author.login : commit.commit.author.name;
}

function authorUrl(commit: Commit) {
  return commit.author?.html_url;
}

export default function CommitsList({ commits }: { commits: Commit[] }) {
  const { locale } = useContext(LocaleContext);

  return (
    <div>
      {commits.map((commit) => {
        const url = authorUrl(commit);

        return (
          <article className={styles.commit} key={commit.sha}>
            <h2 className={styles.message}>
              <a
                className={styles.commitLink}
                href={commit.html_url}
                target="_blank"
                rel="noopener"
              >
                {commit.commit.message}
              </a>
            </h2>

            <p className={styles.committer}>
              <img className={styles.avatar} src={avatarUrl(commit)} alt="" />

              {url && (
                <a href={url} className={styles.committerLink}>
                  {authorName(commit)}
                </a>
              )}

              {!url && (
                <span className={styles.committerLink}>
                  {authorName(commit)}
                </span>
              )}

              <span>{timestamp(locale, commit.commit.author.date)}</span>
            </p>
          </article>
        );
      })}
    </div>
  );
}
