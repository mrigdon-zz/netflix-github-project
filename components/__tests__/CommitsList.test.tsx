import React from "react";
import CommitsList from "../CommitsList";
import { createRenderer } from "react-test-renderer/shallow";

describe("CommitsList", () => {
  it("renders the commits", async () => {
    const renderer = createRenderer();
    renderer.render(
      <CommitsList
        commits={[
          {
            sha: "12345",
            html_url: "https://github.com/netflix/eureka/commits/12345",
            commit: {
              message: "Merged PR #42",
              committer: {
                avatar_url: "/avatar.jpg",
              },
              author: {
                name: "Joe Smith",
                date: "2021-09-09T14:57:48.345Z",
              },
            },
          },
        ]}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
