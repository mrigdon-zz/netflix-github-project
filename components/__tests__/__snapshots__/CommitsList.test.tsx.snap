// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitsList renders the commits 1`] = `
<div>
  <article
    className="commit"
  >
    <h2
      className="message"
    >
      <a
        className="commitLink"
        href="https://github.com/netflix/eureka/commits/12345"
        rel="noopener"
        target="_blank"
      >
        Merged PR #42
      </a>
    </h2>
    <p
      className="committer"
    >
      <img
        alt=""
        className="avatar"
        src="/avatar.jpg"
      />
      <span
        className="committerLink"
      >
        Joe Smith
      </span>
      <span>
        Sep 9, 2021, 7:57 AM
      </span>
    </p>
  </article>
</div>
`;
